### PR TITLE
Refactor py2 lazyproperty

### PR DESCRIPTION
**Pull request by ernestoarbitrio**
Lazyproperty decorator refactoring using builtin `functools` lib

Linked issue : https://github.com/histolab/histolab/issues/159

### CODE REVIEW

**Review by alessiamarcolini — APPROVED** [review]
LGTM

File: src/histolab/tiler.py [hunk 1]
```diff
@@ -48,7 +43,6 @@ class Tiler(Protocol):
     level: int
     tile_size: int
 
-    @lru_cache(maxsize=100)
```
Comment (alessiamarcolini) [inline src/histolab/tiler.py:51]: why did you remove it?

Comment (ernestoarbitrio) [reply src/histolab/tiler.py]: i misunderstood something ... sorry ... restored right now
